archinfo/arch_soot.py

```diff
@@ -40,13 +45,56 @@ def from_string(cls, tstr):
     def fullname(self):
         return "%s.%s" % (self.class_name, self.name)
 
+    def matches_with_native_name(self, native_method):
+        """
+        Name of native methods are getting encoded, s.t. they translate into valid C function names.
+        :return: True, if name of soot method matches the mangled native name.
+        """
+        
+        if "__" in native_method:
+            # if native methods are overloaded, two underscores are used
+            # TODO check function signature
+            raise NotImplementedError('Overloaded native methods are not supported.')
+
+        # demangle native name
+        native_method =  native_method.replace('_1', '_')
+        # TODO unicode escaping
```

Comment:
Just curious, when do you see Unicode strings in method names?

Reply:
Thats a good question, which I don't have an answer for. However, I didn't give it much thought and just added the comment to be complete (since there exist an escape sequence for unicode characters in the JNI specification)